lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -168,6 +217,171 @@ def partition_pdf(
+            stat = os.stat(pdffile.name)
+            assert stat.st_size == data_len
+            return self._partition_pdf_batched_named(
+                pdffile.name,
+                threshold,
+                use_ocr,
+                ocr_images,
+                ocr_tables,
+                extract_table_structure,
+                table_structure_extractor,
+                extract_images,
+                batch_size,
+            )
+
+    def _partition_pdf_batched_named(
+        self,
+        filename: str,
+        threshold: float = 0.4,
+        use_ocr=False,
+        ocr_images=False,
+        ocr_tables=False,
+        extract_table_structure=False,
+        table_structure_extractor=None,
+        extract_images=False,
+        batch_size: int = 1,
+    ) -> List[List["Element"]]:
+        if extract_table_structure and not table_structure_extractor:
+            table_structure_extractor = DEFAULT_TABLE_STRUCTURE_EXTRACTOR(device=self.device)
+
+        pdfminer = None
+        exec = ProcessPoolExecutor(max_workers=1)
+        if not use_ocr:
+            with LogTime("start_pdfminer", log_start=True):
+                pdfminer = exec.submit(self._run_pdfminer, filename)
+
+        deformable_layout = []
+        if tracemalloc.is_tracing():
+            before = tracemalloc.take_snapshot()
+        for i in convert_from_path_streamed_batched(filename, batch_size):
+            parts = self.process_batch(
+                i,
+                threshold=threshold,
+                use_ocr=use_ocr,
+                ocr_images=ocr_images,
+                ocr_tables=ocr_tables,
+                extract_table_structure=extract_table_structure,
+                table_structure_extractor=table_structure_extractor,
+                extract_images=extract_images,
+            )
+            assert len(parts) == len(i)
+            deformable_layout.extend(parts)
+            if tracemalloc.is_tracing():
+                gc.collect()
+                after = tracemalloc.take_snapshot()
+                top_stats = after.compare_to(before, "lineno")
+
+                print("[ Top 10 differences ]")
+                for stat in top_stats[:10]:
+                    print(stat)
+                before = after
+                display_top(after)
+
+        if pdfminer is not None:
+            with LogTime("wait_for_pdfminer", log_start=True):
+                pdfminer_layout = pdfminer.result()
+            assert len(pdfminer_layout) == len(deformable_layout), f"{len(pdfminer_layout)} vs {len(deformable_layout)}"
+            with LogTime("pdfminer_supplement"):
+                for d, p in zip(deformable_layout, pdfminer_layout):
+                    self._supplement_text(d, p)
+
+        if tracemalloc.is_tracing():
+            (current, peak) = tracemalloc.get_traced_memory()
+            logging.info(f"Memory Usage current={current} peak={peak}")
+            top = tracemalloc.take_snapshot()
+            display_top(top)
```

Comment:
It would be cool if it's easy to hide this dirty laundry.

Reply:
Sure.